Page: My Availabilities

Plebysnacc opened the pull request:
Implementation is straight forward, only a simple table update.

If we had something different and more sophisticated in mind, just tell me what to change 👍

Plebysnacc commented:
Idea I had:
- when removing all availabilities: show confirmation dialog.

dheidemann commented on frontend/app/(settings)/profile/availabilities/page.tsx:
```diff
@@ -0,0 +1,131 @@
+  }, [rowSelection])
+
+
+  async function onSubmit() {
```
bit weird behavior by a user, but the following scenario is possible: user deselects events, saves, selects them again. But then cannot save, because the button is disabled. If understood correctly on submit you can just call `setPreviousEventIds(selectedEventIds)` to fix this

Plebysnacc replied:
Yea, that's probably the fix - gonna fix that later

dheidemann commented on frontend/app/(settings)/profile/availabilities/page.tsx:
```diff
@@ -0,0 +1,140 @@
+            setRowSelection={setRowSelection}
+          />
+          <div className={'flex justify-between items-center mt-5 w-full gap-x-12'}>
+            <Button
+              variant={"outline"}
+              className={'flex-grow-[0.25]'}
+              onClick={() => setRowSelection(createRowSelectionFromEventIds(previousEventIds))}
+            >
+              Cancel
+            </Button>
```
for this functionality i am more familiar with a reset button that appears with an icon like this: https://lucide.dev/icons/rotate-ccw

no blocker, just a suggestion if you like this more

Plebysnacc replied:
Yeah that's cool, too. Gonna add it

dheidemann reviewed (changes requested):
unfortunately that didn't fully resolve the issue. If you select, save, deselect everything, the save button is disabled. I would be fine with just leaving the button enabled all the time

Plebysnacc commented:
Gonna try to fix it this week. Not gonna lemme get stopped by some state check

Plebysnacc commented:
Try no. 3!

I changed the button and think I fixed the bug. While working on it I realized that SMTP related errors will break the view, although the user is still updated. So I added really basic error handling

dheidemann reviewed (approved):
hot
